cookbook/case_studies/house_price_prediction/house_price_predictor.py

```diff
@@ -0,0 +1,253 @@
+    _price = gen_price(_house)
+    return [
+        _price,
+        _house["YEAR_BUILT"],
+        _house["SQUARE_FEET"],
+        _house["NUM_BEDROOMS"],
+        _house["NUM_BATHROOMS"],
+        _house["LOT_ACRES"],
+        _house["GARAGE_SPACES"],
+    ]
+
+# %%
+# Dataframe with all the houses details:
+def gen_houses(num_houses) -> pd.DataFrame:
+    _house_list = []
+    for i in range(num_houses):
+        _house_list.append(gen_random_house())
+    _df = pd.DataFrame(
+        _house_list,
+        columns=[
+            "PRICE",
+            "YEAR_BUILT",
+            "SQUARE_FEET",
+            "NUM_BEDROOMS",
+            "NUM_BATHROOMS",
+            "LOT_ACRES",
+            "GARAGE_SPACES",
+        ],
+    )
+    return _df
+
+# %%
+# Splitting the Data:
+def split_data(
+    df: pd.DataFrame, seed: int, split: typing.List[float]
+) -> (np.ndarray, np.ndarray, np.ndarray):
+    # split data into train and test sets
+    seed = seed
+    val_size = split[1]
+    test_size = split[2]
+
+    num_samples = df.shape[0]
+    x1 = df.values[
+        :num_samples, 1:
+    ]  # keep only the features, skip the target, all rows
+    y1 = df.values[:num_samples, :1]  # keep only the target, all rows
+
+    # Use split ratios to divide up into train/val/test
+    x_train, x_val, y_train, y_val = train_test_split(
+        x1, y1, test_size=(test_size + val_size), random_state=seed
+    )
+    # Of the remaining non-training samples, give proper ratio to validation and to test
+    x_test, x_test, y_test, y_test = train_test_split(
+        x_val, y_val, test_size=(test_size / (test_size + val_size)), random_state=seed
+    )
+    # Reassemble the datasets with target in first column and features after that
+    _train = np.concatenate([y_train, x_train], axis=1)
+    _val = np.concatenate([y_val, x_val], axis=1)
+    _test = np.concatenate([y_test, x_test], axis=1)
+
+    return _train, _val, _test
+
+# %%
+# Generating the train, val, and test datasets:
+def generate_data(
+    loc: str, number_of_houses: int, seed: int
+) -> (np.ndarray, np.ndarray, np.ndarray):
+    _houses = gen_houses(number_of_houses)
+    _train, _val, _test = split_data(_houses, seed, split=SPLIT_RATIOS)
+    return _train, _val, _test
+
+# %%
+# Generating CSV files to store train, val, and test datasets:
+def save_to_dir(path: str, n: str, arr: np.ndarray) -> str:
+    d = os.path.join(path, n)
+    os.makedirs(d, exist_ok=True)
+    save_to_file(d, n, arr)
+    return d
+
+
+def save_to_file(path: str, n: str, arr: np.ndarray) -> str:
+    f = f"{n}.csv"
+    f = os.path.join(path, f)
+    np.savetxt(f, arr, delimiter=",", fmt="%.2f")
+    return f
+
+# %%
+# Step 4: Task -- Generating & Splitting the Data
+# -----------------------------------------------
+# Train and validation datasets are directories (consisting of one CSV file) and test data is a CSV file
+@task(cache=True, cache_version="0.1", limits=Resources(mem="600Mi"))
+def generate_and_split_data(
```

Comment:
can this return pandas dataframes itself? and simplify rather than converting to a csv?

Reply:
Done.